src/synth/fluid_synth.h

```diff
@@ -142,22 +142,22 @@ struct _fluid_synth_t
   unsigned int storeid;
   fluid_rvoice_eventhandler_t* eventhandler;
 
-  float reverb_roomsize;             /**< Shadow of reverb roomsize */
-  float reverb_damping;              /**< Shadow of reverb damping */
-  float reverb_width;                /**< Shadow of reverb width */
-  float reverb_level;                /**< Shadow of reverb level */
+  fluid_atomic_float_t reverb_roomsize;             /**< Shadow of reverb roomsize */
+  fluid_atomic_float_t reverb_damping;              /**< Shadow of reverb damping */
+  fluid_atomic_float_t reverb_width;                /**< Shadow of reverb width */
+  fluid_atomic_float_t reverb_level;                /**< Shadow of reverb level */
 
-  int chorus_nr;                     /**< Shadow of chorus number */
-  float chorus_level;                /**< Shadow of chorus level */
-  float chorus_speed;                /**< Shadow of chorus speed */
-  float chorus_depth;                /**< Shadow of chorus depth */
-  int chorus_type;                   /**< Shadow of chorus type */
+  fluid_atomic_int_t chorus_nr;                     /**< Shadow of chorus number */
+  fluid_atomic_float_t chorus_level;                /**< Shadow of chorus level */
+  fluid_atomic_float_t chorus_speed;                /**< Shadow of chorus speed */
+  fluid_atomic_float_t chorus_depth;                /**< Shadow of chorus depth */
+  fluid_atomic_int_t chorus_type;                   /**< Shadow of chorus type */
```

Comment:
I think the chorus_* and reverb_* members of fluid_synth_t can be simple ints, no need for atomics here. Access to those values are protected via the fluid API lock anyway. They are passed by value to the rvoice eventhandler, so the other threads don't access them either.

Reply:
should I make chorus_level, chorus_speed, chorus_depth double type (since they are passed as double in the functions fluid_synth_set_chorus_level, fluid_synth_set_chorus_speed, fluid_synth_set_chorus_depth ?

Reply:
Yes double might be better.

Reply:
dont

Reply:
The `fluid_rvoice_eventhandler_push*` functions take `fluid_real_t` values, not float or double. If FluidSynth is compiled with `-Denable-floats=1`, wouldn't the precision loss cause a warning?
I would suggest defining them as `fluid_real_t` instead.

Reply:
then the API should also change
fluid_synth_set_reverb_full is using double, and the args are directly passed to fluid_rvoice_eventhandler_push5

Reply:
`fluid_rvoice_mixer_set_chorus_params` and `fluid_rvoice_mixer_set_reverb_params` take doubles, while `fluid_rvoice_event_t::realparams` is `fluid_real_t`. I'll take care of that afterwards.

Reply:
There's more:
- fluid_revmodel_set takes floats.
- fluid_revmodel_presets_t uses fluid_real_t
- fluid_revmodel_t uses fluid_real_t

It's all a little... inconsistent :-)

Reply:
Ok, too much for now. Lets postpone this until the new fx_api struct thing. Noted in  #197.